cfgov/unprocessed/js/modules/youtube-api.js

```diff
@@ -0,0 +1,86 @@
+/* ==========================================================================
+   YouTube API
+   ========================================================================== */
+
+import jsLoader from './util/js-loader';
+
+const IMAGE_URL = 'https://img.youtube.com/vi/%video_id%/maxresdefault.jpg';
+const SCRIPT_API = 'https://www.youtube.com/iframe_api';
+
+/**
+ * The "onYouTubeIframeAPIReady" function is automatically called when the
+ * YouTube iFrame API is done loading.
+ * See https://developers.google.com/youtube/iframe_api_reference
+ * @param {Function} callback - function to call when the API is ready.
+ */
+function attachAPIReadyCallback( callback ) {
+  window.onYouTubeIframeAPIReady = callback;
+}
+
+/**
+ * Embed the YouTube iFrame API.
+ * @param {Function} [callback] - function to call when the script is loaded.
+ */
+function embedVideoScript( callback ) {
+  jsLoader.loadScript( SCRIPT_API, callback );
+}
+
+/**
+ * Load Youtube max res image if it exists.
+ * TODO: Replace this method by calling the Youtube data API.
+ * https://developers.google.com/youtube/v3/getting-started#fields
+ */
+/**
+ * Load Youtube max res image if it exists.
+ * TODO: Replace this method by calling the Youtube data API.
+ *       https://developers.google.com/youtube/v3/getting-started#fields
+ * @param {string} videoId - A YouTube video ID.
+ * @returns {string} The image URL.
+ */
+function fetchImageURL( videoId ) {
+  if ( !videoId ) {
+    throw new Error( 'No Video ID provided!' );
+  }
+
+  return IMAGE_URL.replace( '%video_id%', videoId );
+}
+
+/**
+ *
```

Comment:
Could use a short summary of what the function does in this JSDoc.

Reply:
Added